saml/response.go

```diff
@@ -151,6 +152,14 @@ func (sp *ServiceProvider) ParseResponse(
 		}
 	}
 
+	if !opts.skipSignatureValidation {
```

Comment:
Can we make this fully optional? Something like `opts.ResponseSigned` or `opts.ResponseAndAssertionsSigned`?

Reply:
Yes agree that we need to make this optional. 
I had a discussion internally and moved the PR to draft again to bring the optional field change + test the e2e flow with vault.

Reply:
Made all settings optional.